### DISCUSSION TITLE
Connect lines into an open polygon

### DISCUSSION THREAD

**Discussion by SeverinoDaDalt** [ManimCommunity/manim#2731] [Q&A] [answered]
Hey! I am new with Manim, and to test the library, today I tried to animate the Cantor function iterative construction:

https://user-images.githubusercontent.com/97095260/169877908-8d8b03f7-9e1c-4142-b75c-df5b58151da9.mp4

What I don't like about the result is that when the function (in red) transforms from one iteration to the next one, the interpolation of the Transform map is odd. I think the problem has to do with the fact that the vertices of consecutive lines in the function are disconnected.
I have done a little research, and I found nothing. Any idea?

**Comment by behackl** [comment, marked as answer]
Heya! It is a bit easier to help if you share a snippet of how you have been approaching your animation so far, but from what I can see your analysis is spot-on. It looks like you are first trying to transform a `Line` into a `VGroup`, and then the `VGroup` into (larger) `VGroup`s of `Lines`.

`Transform` basically tries to match mobjects onto (sub)mobjects, which explains why in the first played animation the complete line transforms into the first segment, and the two others transform out of "thin air".

One way to make this more natural instead, is to transform between specialized mobjects:
```py
def construct(self):
    initial_line = VMobject(color=RED).set_points_as_corners([[0,0,0], [1, 1, 0]])
    new_line = VMobject(color=RED).set_points_as_corners([[0,0,0], [1/3, 1/2, 0], [2/3, 1/2, 0], [1, 1, 0]])
    self.play(Create(initial_line))
    self.wait()
    self.play(Transform(initial_line, new_line))
    self.wait()
```
There probably is a better way of doing this, depending on how you are currently generating the points of the curve, but it should give you an idea of how to make the transformation look better. :-)

**Reply by SeverinoDaDalt** [reply]
That was exactly what I needed! Result:

https://user-images.githubusercontent.com/97095260/169894567-fc17109b-d50a-42e4-ada5-5d7762fe593b.mp4

And since this would have helped me in creating the script: is there any way to retrieve the corners that created the VMObject, something like `initial_line.corners`?

In any case, thank you very much, this was very helpful
